src/hayhooks/server/utils/mcp_utils.py

```diff
@@ -159,7 +159,15 @@ async def run_pipeline_as_tool(name: str, arguments: dict) -> list["TextContent"
         return [TextContent(text=result, type="text")]
 
     if isinstance(pipeline, AsyncPipeline):
-        result = await pipeline.run_async(data=arguments)
+        # Get include_outputs_from from metadata if available
+        metadata = registry.get_metadata(name) or {}
+        outputs_to_include = metadata.get("include_outputs_from")
+
+        if outputs_to_include is not None:
+            result = await pipeline.run_async(data=arguments, include_outputs_from=outputs_to_include)
+        else:
+            result = await pipeline.run_async(data=arguments)
```

Comment:
nit: in general, I think that several of this if/else blocks can be avoided by using a `kwargs` dict.
Maybe it's just a personal taste

Reply:
Makes sense. The same kwargs patterns is used also elsewhere, so I've aligned them. I've updated also `mcp_utils.py`.